sips/sip-16/sip-016-token-metadata.md

```diff
@@ -0,0 +1,479 @@
+existing token, otherwise the value must be `none`. Appendix A specifies the
+exact meaning of "existing" for each function.
+
+For existing tokens, the inner value of the return value must be a string
+representing a resolvable URI.
+
+The schema of the resolvable URI is not specified and should be a well-known
+schema like `https`, `ar`, `ipfs`, `sia`. A `data` URI is also valid, however,
+the length is limited by this SIP.
+
+If a metadata URI function expects a parameter of type `uint` that identifies a
+token and the resulting strings contain `{id}`, then the `{id}` part must be
+replaced by the identifier in decimal format given in the function call.
+
+The resolved data of the URI must be a JSON blob.
+
+## JSON scheme of Metadata
+
+The JSON blob resolved through the URI must follow the following JSON schema.
+
+If metadata were retrieved by a function call containing a token identifier and
+the string `{id}` exists in any JSON value, it MUST be replaced with the actual
+token id in decimal format, by all client software that follows this standard.
+
+```
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Token Metadata",
+    "type": "object",
+    "required": ["sip", "name"],
+    "properties": {
+        "sip": {
+            "type": "number",
+            "description": "SIP number that defines the JSON schema for metadata. For this SIP, the sip number must be `16`."
+        },
+        "name": {
+            "type": "string",
+            "description": "Identifies the asset which this token represents"
+        },
+        "description": {
+            "type": "string",
+            "description": "Describes the asset which this token represents"
+        },
+        "image": {
+            "type": "string",
+            "description": "A URI pointing to a resource with MIME type image/* representing the asset to which this token represents. Consider making any images at a width between 320 and 1080 pixels and aspect ratio between 1.91:1 and 4:5 inclusive. If the token represents a media file of different MIME type or of higher quality defined in property 'raw_media_file_uri', then this image should be used as preview image like a cover for music, or an low-res image."
+        },
+        "attributes": {
+            "type": "array",
+            "description": "Additional attributes of the token that are \"observable\". See section below. Values may be strings, numbers, object or arrays.",
+            "items": {
+                "type": "object",
+                "required": ["trait_type", "value"],
+                "properties": {
+                    "display_type": {"type": "string"},
+                    "trait_type": {"type": "string"},
+                    "value": {"anyOf": [{"type": "object"}, {"type": "string"}, {"type": "number"}, {"type": "integer"}, {"type": "boolean"}, {"type": "array"}]}
+                }
+            }
+        },
+        "properties": {
+            "type": "object",
```

Comment:
@friedger shouldn't this be an `array` of `object`s (following the schema specified in **Properties**)? Similar to `attributes` above